R/zchunk_LA102.ag_LDS_R_C_GLU.R

```diff
@@ -33,49 +35,88 @@ module_aglu_LA102.ag_LDS_R_C_GLU_DISABLED <- function(command, ...) {
+      # group by GCAM region, GCAM commodity, and GLU for the aggregation:
+      group_by(GCAM_region_ID, GCAM_commodity, GLU) %>%
+      # aggregate to the GCAM region, commodity, GLU level:
+      summarise_if(is.numeric, sum) %>%
+      # convert units from ton (t) to Megatons (Mt):
+      mutate(value=value*CONV_TON_MEGATON) %>%
+      # omit na values, since they do not appear in the original table:
+      na.omit() ->
+      # store in the final, labeled table for production in units Mt at the region-commodity-glu level:
+      L102.ag_Prod_Mt_R_C_GLU
+
 
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L102.ag_HA_bm2_R_C_GLU %>%
+      add_title("Harvested area by GCAM region / commodity / GLU") %>%
+      add_units("Thousand km^2 = billion m^2 (bm2") %>%
+      add_comments("LDS harvested area data, sourced from SAGE, Hyde, and others,") %>%
+      add_comments("is read in at the iso-GLU-GTAP_crop level, and converted to GCAM") %>%
+      add_comments("regions, commodities, and units.") %>%
       add_legacy_name("L102.ag_HA_bm2_R_C_GLU") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("common/iso_GCAM_regID",
+                     "aglu/FAO_ag_items_PRODSTAT",
+                     "L100.LDS_ag_HA_ha",
+                     "L100.LDS_ag_prod_t")  %>%
+      add_flags(FLAG_SUM_TEST) ->
       L102.ag_HA_bm2_R_C_GLU
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L102.ag_Prod_Mt_R_C_GLU %>%
+      add_title("Crop production by GCAM region / commodity / GLU") %>%
+      add_units("Megatons (Mt)") %>%
+      add_comments("LDS production data, sourced from SAGE, Hyde, and others,") %>%
+      add_comments("is read in at the iso-GLU-GTAP_crop level, and converted to GCAM") %>%
+      add_comments("regions, commodities, and units.") %>%
       add_legacy_name("L102.ag_Prod_Mt_R_C_GLU") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("common/iso_GCAM_regID",
+                     "aglu/FAO_ag_items_PRODSTAT",
+                     "L100.LDS_ag_HA_ha",
```

Comment:
Excellent point - I've just been copying the full input files to the precursors at the beginning, but that is not completely accurate. I'll start double checking and updating at the end.